src/cmd/tools/verify_data_files/main/main.go

```diff
@@ -447,11 +415,14 @@ func fixFileSet(
 			return fmt.Errorf("encountered an error not enabled to fix: %v", err)
 		}
 
-		var writeIdentTags ident.Tags
-		writeIdentTags.Reset(writeTags)
+		metadata := persist.NewMetadataFromIDAndTagIterator(id, tagsCopy,
+			persist.MetadataOptions{
+				FinalizeID:          true,
+				FinalizeTagIterator: true,
```

Comment:
Should `FinalizeTags` be on too?

Reply:
Nah, we don't provide Tags here, just TagIterator.